Sources/AngouriMath/Convenience/MathS.cs

```diff
@@ -356,103 +356,104 @@ public static class Hyperbolic
-                => Functions.Series.MultivariableTaylorExpansionTerms(expr, exprToPolyVars);
+            public static Entity Taylor(Entity expr, int degree, params (Variable exprVariable, Variable polyVariable, Entity point)[] exprToPolyVars)
+            => Functions.Series.TaylorExpansion(expr, degree, exprToPolyVars);
 
             /// <summary>
-            /// Finds the symbolic expression of terms of the Taylor expansion of the given function with multiple variables.
-            /// https://en.wikipedia.org/wiki/Taylor_series#Taylor_series_in_several_variables
+            /// Finds the symbolic expression of terms of the Taylor expansion of the given function,
+            /// https://en.wikipedia.org/wiki/Taylor_series
+            /// 
+            /// Do NOT call ToArray() or anything like this on the result of this method. It is 
+            /// infinite iterator. To get a finite result as in a sum of finite number of terms,
+            /// call TaylorExpansion
             /// </summary>
             /// <param name="expr">
-            /// The function, to find Taylor expansion of
+            /// The function to find the Taylor expansion of
             /// </param>
             /// <param name="exprToPolyVars">
-            /// Takes the place of the expression variable, polynomial variable, and the differentiation point for each variable.
-            /// </param>
-            /// <param name="degree">
-            /// The degree of the taylor polynomial desired.
+            /// The variable/s to take the series over, the variable the series will be over,
+            /// and the variable values at which the Taylor polynomial will be found
+            /// (e. g. if you want to find the taylor polynomial of Sin("t") around t=1, and want
+            ///  n to take that place in the series, then you may want to use ("t","n","1") for this argument)
             /// </param>
             /// <returns>
-            /// An expression in the polynomial form over the poly variables given in <paramref name="exprToPolyVars"/>
+            /// An infinite iterator over the terms of Taylor series of the given expression.
             /// </returns>
-            public static Entity MultivariableTaylorExpansion(Entity expr, int degree, params (Variable exprVariable, Variable polyVariable, Entity value)[] exprToPolyVars)
-                => Functions.Series.MultivariableTaylorExpansion(expr, degree, exprToPolyVars);
+            public static IEnumerable<Entity> TaylorTerms(Entity expr, params (Variable exprVariable, Variable polyVariable, Entity point)[] exprToPolyVars)
+            {
+                if (exprToPolyVars.Length == 1)
+                    return Functions.Series.SingleTaylorExpansionTerms(expr, exprToPolyVars[0].exprVariable, exprToPolyVars[0].polyVariable, exprToPolyVars[0].point);
+                else
+                    return Functions.Series.MultivariableTaylorExpansionTerms(expr, exprToPolyVars);
+            }
```

Comment:
Maybe we don't even need `SingleTaylorExpansion` now, what do you think?

Reply:
I'm going to leave it in for now, because I think it's a little faster. The multivariable variant will make space for coefficient and degrees of terms even if they will always be zero for a single variable. If it becomes a maintainability problem in the future, we can always get rid of it.